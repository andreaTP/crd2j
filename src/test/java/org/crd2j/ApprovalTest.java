package org.crd2j;

import io.fabric8.kubernetes.client.DefaultKubernetesClient;
import io.fabric8.kubernetes.client.KubernetesClient;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.Optional;

import static org.assertj.core.api.Assertions.assertThat;

public class ApprovalTest {

    KubernetesClient client = new DefaultKubernetesClient();
    CRGeneratorRunner runner = new CRGeneratorRunner();

    @Test
    void testCrontabCrd() {
        // Arrange
        var crd = client
                .apiextensions()
                .v1()
                .customResourceDefinitions()
                .load(this
                        .getClass()
                        .getClassLoader()
                        .getResource("crontab-crd.yml"))
                .get();

        // Act
        var writables = runner.generate(crd, Optional.empty());

        // Assert
        assertThat(writables.size()).isEqualTo(1);

        var writable = writables.get(0);

        Approvals.verifyAll("CrontabJavaCr", List.of(
            writable.getJavaClass("CronTab"),
            writable.getJavaClass("Spec"),
            writable.getJavaClass("Status"))
        );
    }

    @Test
    void testKeycloakCrd() {
        // Arrange
        var crd = client
                .apiextensions()
                .v1()
                .customResourceDefinitions()
                .load(this
                        .getClass()
                        .getClassLoader()
                        .getResource("keycloak-crd.yml"))
                .get();

        // Act
        var writables = runner.generate(crd, Optional.empty());

        // Assert
        assertThat(writables.size()).isEqualTo(1);

        var writable = writables.get(0);

        Approvals.verifyAll("KeycloakJavaCr", List.of(
                writable.getJavaClass("Keycloak"),
                writable.getJavaClass("Spec"),
                writable.getJavaClass("Status"))
        );
    }

    @Test
    void testAkkaMicroservicesCrd() {
        // Arrange
        var crd = client
                .apiextensions()
                .v1()
                .customResourceDefinitions()
                .load(this
                        .getClass()
                        .getClassLoader()
                        .getResource("akka-microservices-crd.yml"))
                .get();

        // Act
        var writables = runner.generate(crd, Optional.empty());

        // Assert
        assertThat(writables.size()).isEqualTo(1);

        var writable = writables.get(0);

        Approvals.verifyAll("AkkaMicroservicesJavaCr", List.of(
                writable.getJavaClass("AkkaMicroservice"),
                writable.getJavaClass("Spec"),
                writable.getJavaClass("Status"))
        );
    }
}
