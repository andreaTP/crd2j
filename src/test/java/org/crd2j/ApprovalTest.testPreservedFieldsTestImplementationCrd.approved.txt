PreservedFieldsJavaCr[0] = package v1;

public class TestPreservedFields {

    @com.fasterxml.jackson.annotation.JsonAnyGetter()
    @com.fasterxml.jackson.annotation.JsonAnySetter()
    private java.util.Map<String, Object> properties = new java.util.HashMap<String, Object>();

    @com.fasterxml.jackson.annotation.JsonIgnore()
    public java.util.Map<String, Object> getProperties() {
        return properties;
    }

    @com.fasterxml.jackson.annotation.JsonProperty()
    public void setProperties(java.util.Map<String, Object> properties) {
        this.properties = properties;
    }
}
